RadixWallet/Profile/Entity/EntityProtocol.swift

```diff
@@ -2,6 +2,8 @@ import EngineToolkit
 
 // MARK: - EntityBaseProtocol
 public protocol EntityBaseProtocol {
+	typealias Flags = IdentifiedArrayOf<EntityFlag>
```

Comment:
I think `OrderedSet` is preferable over `IdentifiedArray` is it not? We use Identified when it is a struct with an ID right? When the ID *is* the `value` then it is the same thing as `OrderedSet`, just more confusing to user IdentifiedArrayOf? Right?

Reply:
I maintain that a flags variable should be a Set, or better yet, an OptionSet, but that might be overkill.

- When you only care about inclusion vs exclusion, and it should be unique:
Set or OptionSet (the latter gives some syntactic sugar)

- When you only care about order:
Array

- When you only care about order but the elements have to be identified because of ForEach
still just Array

- When you care about order, but also want uniqueness of elements (and O(1) lookup):
OrderedSet

- When you want explicit keys and ignore order (and want O(1) ops)
Dictionary

- When you want explicit keys and order (and want O(1) ops)
OrderedDictionary

- When you want implicit keys and order (and want O(1) ops)
IdentifiedArray

Reply:
We cannot use Sets as we do have to be compatible with Android encoding/decoding, where:
`flags: Set<EntityFlag>` is encoded as `"flags":["deletedByUser"]`.

Swift encoder is different, where `Set<EntityFlag>` or `OrderedSet<EntityFlag>` is encoded as:
```
"flags": {
    {
       "deletedByUser": {}
    }
}
``

Reply:
We could always do it manually, but it doesn't seem worth it. So sets are encoded as dictionaries with the elements as keys, and void as values?

Reply:
> So sets are encoded as dictionaries with the elements as keys, and void as values?
Yes, so it seems.

Reply:
It seems Set is encoded and decoded just like Array in Swift, try this:
```Swift
	struct TestStruct: Codable {
		let set: Set<String>
		let array: [String]
	}

	func test_encode() throws {
		let value = TestStruct(set: ["one", "two"], array: ["three", "four"])
		let encoded = try JSONEncoder().encode(value)
		print("ENCODED:\n\(String(data: encoded, encoding: .utf8))")
	}

	func test_decode() throws {
		let string = "{\"array\":[\"three\",\"four\"],\"set\":[\"one\",\"two\"]}"
		guard let data = string.data(using: .utf8) else { return }
		let decoded = try JSONDecoder().decode(TestStruct.self, from: data)
		print("DECODED:\n\(decoded)")
	}
```

Reply:
Works even for OrderedSet

Reply:
Ah, that's surprising, but I guess enums are hard to encode for swift